src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -116,15 +110,24 @@ private <T> List<T> runSnapshotBuilderQuery(
 
   public SnapshotBuilderGetConceptsResponse getConceptChildren(
       UUID datasetId, Integer conceptId, AuthenticatedUserRequest userRequest) {
-    // TODO: Build real query - this should get the name and ID from the concept table, the count
-    // from the occurrence table, and the existence of children from the concept_ancestor table.
     Dataset dataset = datasetService.retrieve(datasetId);
-
-    String cloudSpecificSQL = buildConceptChildrenQuery(dataset, conceptId, userRequest);
-
+    String cloudSpecificSQL;
+    var cloudPlatformWrapper = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+    if (cloudPlatformWrapper.isGcp()) {
+      DatasetModel datasetModel = datasetService.retrieveDatasetModel(datasetId, userRequest);
+      cloudSpecificSQL =
+          buildConceptChildrenQuery(conceptId, BigQueryVisitor.bqTableName(datasetModel));
+    } else if (cloudPlatformWrapper.isAzure()) {
+      String datasourceName =
+          datasetService.getOrCreateExternalAzureDataSource(dataset, userRequest);
+      cloudSpecificSQL =
+          buildConceptChildrenQuery(conceptId, SynapseVisitor.azureTableName(datasourceName));
+    } else {
+      throw new NotImplementedException("Cloud platform not implemented");
+    }
```

Comment:
Could we add this logic to runSnapshotBuilderQuery() and passing buildConceptChildrenQuery as a function parameter?

Reply:
I can definitely move this logic into runSnapshotBuilderQuery(). Let me know what you think about this new approach. I didn't pass buildConceptChildrenQuery as a function parameter because it takes different arguments depending on if its for bq or azure. So unless I'm missing something, I think I'd have to hand runSnapshotBuilderQuery() both versions of the function call so I'd need to compute both.

Reply:
But now that I added all that its not a generic method anymore. So I need to do something else.

Reply:
Yeah, I agree. Thanks for looking into the other option! I think this looks good.